app/models/client.rb

```diff
@@ -1,45 +1,57 @@
-  attribute :created,  DateTime,  mapping: { type: :date }
-  attribute :name,  String,  mapping: { type: 'text' }
-  attribute :contact_name,  String, default: "", mapping: { type: 'text' }
-  attribute :contact_email,  String,  mapping: { type: 'keyword' }
-  attribute :re3data,  String,  mapping: { type: 'keyword' }
-  attribute :doi_quota_allowed,  Integer, default: 0, mapping: { type: 'integer' }
-  attribute :version,    Integer, default: 0, mapping: { type: 'integer' }
-  attribute :role_name,  String, default: "ROLE_ALLOCATOR" , mapping: { type: 'keyword' }
-  attribute :is_active,  String, default: "\x01", mapping: { type: 'boolean' }
-  attribute :doi_quota_used,  Integer, default: -1, mapping: { type: 'integer' }
-  attribute :comments,  String,  mapping: { type: 'text' }
-  attribute :domains,  String,  mapping: { type: 'text' }
-  attribute :password,  String,  mapping: { type: 'text' }
-  attribute :provider_id,  String,  mapping: { type: 'keyword' }
-  attribute :provider_symbol,  String,  mapping: { type: 'keyword' }
-  attribute :experiments,  String,  mapping: { type: 'text' }
-  attribute :url,  String,  mapping: { type: 'text' }
-  attribute :deleted_at,  Date,  mapping: { type: 'date' }
-  attribute :prefixes,  String,  mapping: { type: 'text' }
-
-
-  validates :symbol, :name,  :contact_email, presence: :true
+  include Importable
+
+  index_name "clients-#{Rails.env}"
+
+  attribute :symbol, String, mapping: { type: 'keyword' }
+  attribute :year, Integer, mapping: { type: 'integer' }
+  attribute :created, DateTime, mapping: { type: :date }
+  attribute :name, String, mapping: { type: 'keyword' }
+  attribute :contact_name, String, default: "", mapping: { type: 'text' }
+  attribute :contact_email, String, mapping: { type: 'keyword' }
+  attribute :re3data, String, mapping: { type: 'keyword' }
+  attribute :version, Integer, default: 0, mapping: { type: 'integer' }
+  attribute :is_active, Integer, default: true, mapping: { type: 'boolean' }
+  attribute :domains, String, mapping: { type: 'text' }
+  attribute :provider_id, String, mapping: { type: 'keyword' }
+  attribute :url, String, mapping: { type: 'text' }
+  attribute :deleted_at, Date, mapping: { type: 'date' }
+  attribute :prefixes, String, mapping: { type: 'text' }
+
+  validates :symbol, :name, :contact_name, :contact_email, presence: :true
   # validates :symbol, symbol: {uniqueness: true} # {message: "This Client ID has already been taken"}
-  validates :contact_email, format:  {  with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i }
-
-
-  before_create :instance_validations
+  validates_format_of :contact_email, :with => /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i, message: "contact_email should be an email"
+
+  #before_create :instance_validations
+
+  def self.query(query, options={})
+    search({
+      from: options[:from],
+      size: options[:size],
+      sort: [options[:sort], "_doc"],
+      query: {
+        query_string: {
```

Comment:
One of the issues I had with `query_string` is that it doesn't give exact results. That is one reason I didn't use it for filtering. So for filtering I always use **term queries** because that gives you exact results. Otherwise I would think the counts will not be perfect matches @mfenner